### PR TITLE
Add command-line interface

### PR DESCRIPTION
**Pull request by ltalirz**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by ltalirz** [comment]
@kjappelbaum this can be reviewed after the commit bot on the eqeq repo has been convinced to do its job

File: mofchecker/__init__.py [hunk 1]
```diff
@@ -51,6 +51,32 @@
 __all__ = ["__version__", "MOFChecker"]
 
 
+DESCRIPTORS = [
```
Comment (kjappelbaum) [inline mofchecker/__init__.py:54]: I think I wouldn't like to duplicate this. I'll try to address this when addressing the graph initialization problem you mentioned as it is related. I see that you already replaced the function that runs all check with looping over this list, but we still then have this list and the checks dictionary.

File: mofchecker/checks/local_structure/false_oxo.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (kjappelbaum) [inline mofchecker/checks/local_structure/false_oxo.py:1]:
```suggestion
```

**Review by kjappelbaum — COMMENTED** [review]
Thanks for bringing back the CLI. I removed it in the recent refactor effort but the one you have here seems easy enough to maintain and pretty useful.

File: mofchecker/__init__.py (hunk 1 above)
Comment (ltalirz) [reply mofchecker/__init__.py:54]: indeed, so the duplication was already there (and I just didn't reduce it).

I'm also wondering a bit whether these different ways of calling things (overvalent, overcoordinated) are really necessary but I guess in the end it does not really hurt.

**Comment by kjappelbaum** [comment]
lgtm!

**Comment by ltalirz** [comment]
There are a large number of pre-commit failures that have nothing to do with this PR.
What is your policy there - do you simply ignore them?

If I may, I would make two changes:
 * lock down the pylint version so that new PRs don't fail because of a pylint update
 * add rules to the ignore list that you don't care about (let me know which those are)

**Comment by ltalirz** [comment]
I'm fixing a couple of pylint issues but not all

**Comment by kjappelbaum** [comment]
thanks, w.r.t. pylint: of course, the policy is not to ignore them (once this round of major refactorings is done). But for some of the more recent commits i indeed ignored them to get some logic changes pushed without focussing to much on the cleanup as there are still a few things that need to move around.

so thanks for looking into them. i'll go through the rest of them soonish

**Pull request merged**